app/controllers/users_controller.rb

```diff
@@ -81,9 +96,19 @@ def user_params
     )
   end
 
-  def remove_linked_convicts(user)
-    return if %w[cpip dpip].include? user.role
+  def remove_linked_convicts(user, mutation: false)
+    return if %w[cpip dpip].include?(user.role) && !mutation
 
     user.convicts.each { |c| Convict.update(c.id, user_id: nil) }
   end
+
+  def removed_linked_appointments(user)
+    future_appointments = user.appointments.joins(:slot).where('slots.date > ?', Date.today)
```

Comment:
Time.zone.today pour éviter de potentiels problèmes de Timezone

--> https://stackoverflow.com/questions/1937161/is-time-zone-now-to-date-equivalent-to-date-today